internal/services/datadog/datadog_monitor_tag_rule_resource.go

```diff
@@ -145,15 +144,6 @@ func resourceDatadogTagRulesCreate(d *pluginsdk.ResourceData, meta interface{})
 	}
 
 	id := rules.NewTagRuleID(monitorId.SubscriptionId, monitorId.ResourceGroupName, monitorId.MonitorName, d.Get("name").(string))
-	existing, err := client.TagRulesGet(ctx, id)
-	if err != nil {
-		if !response.WasNotFound(existing.HttpResponse) {
-			return fmt.Errorf("checking for an existing %s: %+v", id, err)
-		}
-	}
-	if !response.WasNotFound(existing.HttpResponse) {
-		return tf.ImportAsExistsError("azurerm_datadog_monitor_tag_rule", id.ID())
-	}
```

Comment:
This is a core behavioural feature of the provider, so we shouldn't be requiring the `requiredImports` check, instead we can update it to confirm if this is configured with the default settings, and raise the `requiredImports` error if so - can we update this?

Reply:
@tombuildsstuff , seems service API only support name "default" for datadog monitor tag rule.

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/19754191/d73e8958-1f66-496b-a602-0b20311c2f86)

Reply:
I updated the code to check if the "Tag Rule" exists or the "Tag Rule" is with default settings.

Service team confirmed that below behaviors are expected.
1. Datadog Monitor Tag Rule only allows name "default".
2. For the completely new Datadog Monitor, there is no existing Datadog Monitor Tag Rule under it.
3. For the recreated Datadog Monitor with same name, there always is an existing default Datadog Monitor Tag Rule under it.